extensions/standard-processors/processors/HashContent.h

```diff
@@ -51,10 +51,15 @@ namespace { // NOLINT
     ret_val.second = 0;
     std::array<std::byte, HASH_BUFFER_SIZE> buffer{};
     EVP_MD_CTX *context = EVP_MD_CTX_new();
-    const auto guard = gsl::finally([&context]() {
+    EVP_MD *md5 = EVP_MD_fetch(nullptr, "MD5", "-fips");
+    const auto guard = gsl::finally([&context, &md5]() {
+      EVP_MD_free(md5);
       EVP_MD_CTX_free(context);
     });
-    EVP_DigestInit_ex(context, EVP_md5(), nullptr);
+    if (!md5) {
+      return ret_val;
```

Comment:
Should we set `ret_val.second` to a negative number to indicate an error? As it is, we are returning an empty string as the hash without any indication that something is wrong.

Reply:
I think currently it's the same case for other error cases as well like when `session.read` fails, it also returns an empty string for the hash for all hashing algorithms. We can revise the error handling for the hashing overall to indicate the error better, but I think that's out of the scope of this PR.